internal/pkg/term/selector/selector.go

```diff
@@ -264,46 +265,79 @@ func (s *DeploySelect) DeployedService(prompt, help string, app string, opts ...
 
 // Service fetches all services in the workspace and then prompts the user to select one.
 func (s *WorkspaceSelect) Service(msg, help string) (string, error) {
-	serviceNames, err := s.retrieveWorkspaceServices()
+	wsServiceNames, err := s.retrieveWorkspaceServices()
 	if err != nil {
-		return "", fmt.Errorf("list services: %w", err)
+		return "", fmt.Errorf("retrieve services from workspace: %w", err)
 	}
+	summary, err := s.wlLister.Summary()
+	if err != nil {
+		return "", fmt.Errorf("retrieve app name from workspace: %w", err)
+	}
```

Comment:
What do you think about retrieving the summary in the constructor? This way we won't have to invoke the method if `Service` or `Job` is called multiple times with a workspaceSelector (although unlikely)

Reply:
Done. Thanks!